### PR TITLE
[Functions] Various housecleaning

### PR DESCRIPTION
**Pull request by justinkaseman**
### Changes
- FunctionsBilling.sol remove unused maxCallbackGasLimit
- FunctionsClient.sol handleOracleFulfillment emits RequestFulfilled
- FunctionsRequest.sol remove unused requestSignature
- FunctionsBilling.sol gas price is in wei, change variable name
- FunctionsRouter.sol prevent duplicate requestIds from Coordinator

### CODE REVIEW

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol
Comment (bolekk) [inline contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol:23]: comment should say "1 million gwei"

**Review by bolekk — COMMENTED** [review]
LGTM, I'll let other reviewers approve.

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol
Comment (KuphJr) [reply contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol:23]: Actually, I think this is correct.  We switched from gwei to wei for the units here. Right @justinkaseman ?

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsClient.sol
Comment (KuphJr) [inline contracts/src/v0.8/functions/dev/1_0_0/FunctionsClient.sol:60]: nit: I wonder if it makes sense to also emit the response & error?

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol
Comment (KuphJr) [inline contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol:248]: Is this just a sanity check in case, by some incredibly unlucky chance, we encounter a collision?

**Review by KuphJr — DISMISSED** [review]
Couple comments, but LGTM

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol
Comment (bolekk) [reply contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol:23]: This number represents 1 quadrillion wei or 1 million gwei so the comment is incorrect.

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol
Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol:248]: That or a malicious Coordinator passes an intentional duplicate. Just something that should never happen.

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol
Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol:23]: @bolekk  is correct, this is gwei

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol
Comment (RensR) [inline contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol:23]: Since it's a constant, wouldn't uint256 make sense?

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol
Comment (bolekk) [inline contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol:143]: Does this need a corresponding change in tooling?

**Review by bolekk — APPROVED** [review]

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol
Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol:143]: I wouldn't think so, because the variable name is only used within the function.
I believe even if the ABI has the old name as long as the type is the same it will be fine

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsClient.sol
Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/1_0_0/FunctionsClient.sol:60]: It'll cost a teeny bit more gas, would this be valuable to users?
I don't have a strong opinon. If you do I can add, or we could run it by Dev Rel

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol
Comment (KuphJr) [reply contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol:143]: Yes, I will need to fix this in the Functions Toolkit & HH starter kit, but I am already tracking.

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsClient.sol
Comment (KuphJr) [reply contracts/src/v0.8/functions/dev/1_0_0/FunctionsClient.sol:60]: That's fine. Not worth holding this PR.

File: contracts/src/v0.8/functions/tests/1_0_0/FunctionsCoordinator.t.sol
Comment (KuphJr) [inline contracts/src/v0.8/functions/tests/1_0_0/FunctionsCoordinator.t.sol:152]: This test should be checking the returned value against an expected value.  I think this is important in case there is some future change that somehow impacts cost estimates.  Plus, we should make sure the returned cost passes the "sniff test"

File: contracts/src/v0.8/functions/tests/1_0_0/FunctionsCoordinator.t.sol
Comment (KuphJr) [inline contracts/src/v0.8/functions/tests/1_0_0/FunctionsCoordinator.t.sol:170]: same here. Check returned value

File: contracts/src/v0.8/functions/tests/1_0_0/FunctionsCoordinator.t.sol
Comment (justinkaseman) [reply contracts/src/v0.8/functions/tests/1_0_0/FunctionsCoordinator.t.sol:152]: Check returned value against what? The actual charged cost?

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol
Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol:143]: Oh I see what you meant. Yeah, as Morgan said we have the change planned.

File: contracts/src/v0.8/functions/tests/1_0_0/FunctionsCoordinator.t.sol
Comment (KuphJr) [reply contracts/src/v0.8/functions/tests/1_0_0/FunctionsCoordinator.t.sol:170]: The second option is best, but IMO I would be fine with just getting the current value, making sure it makes sense and then hardcoding that as the expected value.

Comment (justinkaseman) [reply contracts/src/v0.8/functions/tests/1_0_0/FunctionsCoordinator.t.sol]: Added a value snapshot

**Review by KuphJr — DISMISSED** [review]
A possible extension in the future might be more tests around cost estimates for varying callback gas amounts (ie: 0, up to the max), but I think that is outside of the scope of this PR.
Approved!

**Review by bolekk — APPROVED** [review]

**Review by KuphJr — APPROVED** [review]

**Pull request merged**
